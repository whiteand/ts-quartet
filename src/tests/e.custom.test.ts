import { e } from "..";
import { testValidatorImpure } from "./testValidatorImpure";

describe("e.custom", () => {
  test("e(e.arrayOf(e.custom(e(e.number))))", () => {
    const checkNumber = e(e.number);
    const checkArrNumber = e(e.arrayOf(e.custom(checkNumber)));
    testValidatorImpure(
      checkArrNumber,
      [[], [1], [1, 2, 3]],
      [null, false, { length: 1, 0: 1 }, ["1"]]
    );
  });
});
