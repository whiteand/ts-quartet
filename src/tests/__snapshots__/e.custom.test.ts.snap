// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e.custom e(e.arrayOf(e.custom(e(e.number)))) 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "elementSchema": Object {
        "customValidator": [Function],
        "innerExplanations": Array [],
        "type": "Custom",
      },
      "type": "ArrayOf",
    },
    "value": null,
  },
]
`;

exports[`e.custom e(e.arrayOf(e.custom(e(e.number)))) 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "elementSchema": Object {
        "customValidator": [Function],
        "innerExplanations": Array [],
        "type": "Custom",
      },
      "type": "ArrayOf",
    },
    "value": false,
  },
]
`;

exports[`e.custom e(e.arrayOf(e.custom(e(e.number)))) 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "elementSchema": Object {
        "customValidator": [Function],
        "innerExplanations": Array [],
        "type": "Custom",
      },
      "type": "ArrayOf",
    },
    "value": Object {
      "0": 1,
      "length": 1,
    },
  },
]
`;

exports[`e.custom e(e.arrayOf(e.custom(e(e.number)))) 4`] = `
Array [
  Object {
    "path": Array [
      0,
    ],
    "schema": Object {
      "customValidator": [Function],
      "innerExplanations": Array [
        Object {
          "path": Array [],
          "schema": Object {
            "type": "Number",
          },
          "value": "1",
        },
      ],
      "type": "Custom",
    },
    "value": "1",
  },
]
`;
